.github/workflows/release.yml

```diff
@@ -0,0 +1,51 @@
+name: envattrs release workflow
+
+on:
+  release:
+    types: [released] # only triggered on release, not pre-release or draft release
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    name: Release to PyPI
+    runs-on: ubuntu-latest
+    env:
+      TARGET_BRANCH: ${{ github.event.release.target_commitish }}
+      DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+      TAG_NAME: ${{ GITHUB_REF_NAME }}
+    steps:
+      - name: Check whether release target branch is default branch
+        if: ${{ TARGET_BRANCH != DEFAULT_BRANCH }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Target branch (${{ TARGET_BRANCH }}) is not default branch (${{ DEFAULT_BRANCH }}).')
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+      - run: |
+          curl -sSL https://install.python-poetry.org | python -
+        env:
+          POETRY_VERSION: 1.3.1
+        shell: bash
+      - run: |
+          echo "PKG_VER=$(poetry version --short)" >> $GITHUB_ENV
+        shell: bash
+      - name: Check whether current tag matches package version
+        if: ${{ TAG_NAME != PKG_VER }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Tag name (${{ TAG_NAME }}) does not match package version (${{ PKG_VER }}).')
+      - name: Release
+        run: poetry publish --build
```

Comment:
will this upload a wheel, an sdist or both?

Reply:
Both, as can be seen here:
https://test.pypi.org/project/envattrs/#files

And mentioned in the docs as well:
https://python-poetry.org/docs/cli/#build